src/network/networkd-link.c

```diff
@@ -689,7 +689,9 @@ static Address* link_find_dhcp_server_address(Link *link) {
 static int link_enter_configured(Link *link) {
         assert(link);
         assert(link->network);
-        assert(link->state == LINK_STATE_SETTING_ROUTES);
+
+        if (link->state != LINK_STATE_SETTING_ROUTES)
+                return 1;
```

Comment:
Actually we have diffrent places with same behaviour I chose the  wrong example.

`link_initialized_and_synced`

```
    if (link->state != LINK_STATE_PENDING)
                return 1

```

But yes we can make `link_enter_configured` void